cosmos/scripts/chaos/cosmosdb_client_network_chaos_tool/cosmosdb_client_network_chaos_tool.ps1

```diff
@@ -0,0 +1,259 @@
+    if ($commaSeparatedPkid)
+    {
+        $commaSeparatedPkid += "," + $partitionKeyRange.id
+    }
+    else
+    {
+        $commaSeparatedPkid += $partitionKeyRange.id
+    }
+}
+
+# Get IP addresses and ports of the gateway and backend nodes
+$addressesResponseJson = & .\get_addresses.ps1 -Endpoint $cosmosDBEndpoint -AccessToken $dataPlaneAccessToken -MasterKey $cosmosDBMasterKey -PartitionKeyIds $commaSeparatedPkid -DatabaseID $databaseId -ContainerId $containerId
+$addressesResponse = $addressesResponseJson | ConvertFrom-Json
+$addresses = $addressesResponse.Addresss
+$backendUriList = New-Object System.Collections.Generic.List[uri]
+foreach ($address in $addresses)
+{
+    $backendUriList += [uri]$address.physcialUri
+}
+$backendUriList = $backendUriList | Select-Object -uniq
+
+$endpointUri = [uri]$cosmosDBEndpoint
+$endpointPort = $endpointUri.Port
+$endpointHost = $endpointUri.Host
+$endpointIpAddress = (Resolve-DnsName $endpointHost).IPAddress
+$subnetMask = "255.255.255.255"
+
+# Adding gateway endpoint for filtering
+$filterString = "[{\`"portHigh\`":$endpointPort,\`"subnetMask\`":\`"$subnetMask\`",\`"address\`":\`"$endpointIpAddress\`",\`"portLow\`":$endpointPort}"
+
+#HashSet to store the unique IP addresses
+$uniqueIPAddressHashSet = New-Object System.Collections.Generic.HashSet[string]
+
+foreach ($backendUri in $backendUriList)
+{
+    $backendHost = $backendUri.Host
+    $ipAddress = (Resolve-DnsName $backendHost).IPAddress
+    $uniqueIPAddressHashSet.Add($ipAddress)
+}
+
+# Adding backend nodes for filtering
+foreach ($ipAddress in $uniqueIPAddressHashSet)
+{
+    $lowPort = 0
+    $highPort = 65535
+    if ($filterString)
+    {
+        $filterString += ",{\`"portHigh\`":\`"$highPort\`",\`"subnetMask\`":\`"$subnetMask\`",\`"address\`":\`"$ipAddress\`",\`"portLow\`":\`"$lowPort\`"}"
+    }
+    else
+    {
+        $filterString += "[{\`"portHigh\`":\`"$highPort\`",\`"subnetMask\`":\`"$subnetMask\`",\`"address\`":\`"$ipAddress\`",\`"portLow\`":\`"$lowPort\`"}"
+    }
+}
+
+if ($filterString)
+{
+    $filterString += "]"
+}    
+
+# Create the experiment json
+$experimentJSON = & .\create_experiment_json.ps1 -FilterString $filterString -DurationOfFaultInMinutes $durationOfFaultInMinutes -FaultRegion $faultRegion -ExperimentName $chaosExperimentName -ResourceGroup $chaosStudioResourceGroupName -SubscriptionId $chaosStudioSubscriptionId -DelayInMs $delayInMs -TargetVMSubRGNameList $targetVMSubRGNameList -TargetVMSSSubRGName $targetVMSSSubRGName -VmssInstanceIdList $vmssInstanceIdList -ChaosExperimentManagedIdentityName $chaosExperimentManagedIdentityName
+
+# Create the experiment on Chaos Studio and Start the experiment
+& .\experiment_operations.ps1 -ExperimentSubscriptionId $chaosStudioSubscriptionId -ExperimentResourceGroup $chaosStudioResourceGroupName  -ExperimentName $chaosExperimentName -ExperimentJSON $experimentJSON -ChaosStudioManagedIdentityClientId $chaosStudioManagedIdentityClientId
+
```

Comment:
There is no exception handling implemented, what is plan regarding that.  It would be great if we can add some retry logic. The exception handling and retry logic comment applies to all the applicable scripts.

Reply:
Seems like I missed adding exception handling to this particular script as I added exception handling to rest of the scripts.

Reply:
Added exception handling and transcription to output the logs to output file created per execution

Reply:
Added retry to all calls that can fail transiently..